core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -64,7 +64,10 @@ class SnapshotService(protected val userInfo: UserInfo, val dataSource: SlickDat
               case t: Throwable =>
                 //fire and forget these undos, we've made our best effort to fix things at this point
                 for {
-                  _ <- deltaLayer.deleteDataset(workspaceContext).unsafeToFuture()
+                  // TODO: AS-724 is it worth deleting the companion and its ref here? If the user attempted to add a snapref,
+                  // they are likely to do so again, and we can just reuse whatever companion was already created.
+                  // This will prevent continual creation/deletion of the same-named dataset.
+                  _ <- deltaLayer.deleteDataset(workspaceContext)
```

Comment:
I would like to consider simplifying SnapshotService by not creating a WSM reference for the companion dataset, and then removing all the accompanying error-handling cases and pet-retrieval. BUT, I'd like to consider that for a future PR and leave it as-is in this one to keep scope down. For that potential future PR … does this sound like a good idea?

Reply:
Are you suggesting simplifying the class by moving that logic elsewhere or by removing the code entirely? I am assuming the latter. I think that's probably fine. It kinda felt like this was just using WSM for the sake of using WSM

Reply:
the latter - don't create the WSM reference-to-companion-dataset at all. I don't think we need it, now that we have one companion per workspace.

Reply:
I'm on board with that 👍